Add Jekyll Scholar Gem and Bibliography Support

mr-pranay opened the pull request:
Modifications regarding publications

svkeerthy commented on _layouts/bibtemplate.html:
```diff
@@ -1,6 +1,289 @@
+
+</div>
+
+
```
Please cleanup the empty lines and commented portion of the code.

svkeerthy commented:
Please write the objective of the PR precisely in the PR title and description.

mr-pranay commented:
- Added Jekyll Scholar gem 
- Created a new folder (_bibliography) to store BibTeX files containing publication information
-  Added a BibTeX template formatting file for rendering publications according to our design requirements
